src/App.tsx

```diff
@@ -9,39 +11,35 @@
 import { PostDetails } from './components/PostDetails';
 import { UserSelector } from './components/UserSelector';
 import { Loader } from './components/Loader';
-import { getUserPosts } from './api/posts';
 import { User } from './types/User';
 import { Post } from './types/Post';
+import { loadUsers } from './features/usersSlice';
+import { fetchUserPosts } from './features/postsSlice';
+import { RootState, AppDispatch } from './app/store';
 
 export const App: React.FC = () => {
-  const [posts, setPosts] = useState<Post[]>([]);
-  const [loaded, setLoaded] = useState(false);
-  const [hasError, setError] = useState(false);
+  const dispatch = useDispatch<AppDispatch>();
+  const users = useSelector((state: RootState) => state.users.items);
+  const [selectedUser, setSelectedUser] = useState<User | null>(null);
 
-  const [author, setAuthor] = useState<User | null>(null);
+  const {
+    items: posts,
+    loaded: postsLoaded,
+    hasError: postsError
+  } = useSelector((state: RootState) => state.posts);
   const [selectedPost, setSelectedPost] = useState<Post | null>(null);
 
-  function loadUserPosts(userId: number) {
-    setLoaded(false);
-
-    getUserPosts(userId)
-      .then(setPosts)
-      .catch(() => setError(true))
-      // We disable the spinner in any case
-      .finally(() => setLoaded(true));
-  }
+  useEffect(() => {
+    dispatch(loadUsers());
+  }, [dispatch]);
```

Comment:
The `useEffect` hook with an empty dependency array will only run once after the initial render. This is correct for loading users initially, but ensure that `loadUsers` does not need to be called again during the component's lifecycle.